cfgrammar/src/lib/yacc/firsts.rs

```diff
@@ -310,6 +315,7 @@ mod test {
         let grm = YaccGrammar::new(
             YaccKind::Original(YaccOriginalActionKind::GenericParseTree),
             "
+          %expect-unused G
```

Comment:
This one shows some of the limitations of the current mechanism for finding unused rules.

It only triggers on `G`, while `C`, `D`, and `F`,  are only used by `G`.

Reply:
So I updated the mechanism to a recursive one, which catches a lot more things.

I went and ignored any *token* also specified in an `expect-unused`, which seemed like an improvement.

So we don't have to `%expect-unused Unmatched "UNMATCHED"` just the rule.
It was less clear to me whether `%expect-unused` should also expect that any subsequent rules can also be unused, so I left a comment there.

So I think this should be fixed in [903f5a4](https://github.com/softdevteam/grmtools/pull/345/commits/903f5a4d1c0bf4e1c4e0211c8fdf9dbf47ce3428)